src/main/kotlin/com/hibob/academy/service/OwnerService.kt

```diff
@@ -1,10 +1,50 @@
 package com.hibob.academy.service
 
 import com.hibob.academy.dao.OwnerDao
+import com.hibob.academy.entity.Owner
+import jakarta.ws.rs.BadRequestException
 import org.springframework.stereotype.Service
+import java.util.*
 
 @Service
 class OwnerService(
     private val ownerDao: OwnerDao
 ) {
+
+    fun getAllOwnersByCompanyId(companyId: Long) =
+        ownerDao.getAllOwnersByCompanyId(companyId)
+
+    fun createOwner(owner: Owner): Owner {
+
+        val generatedId = UUID.randomUUID().mostSignificantBits and Long.MAX_VALUE
```

Comment:
Your table can generate the id - and you can get the id back from the ownerDao.createOwner